packages/rule-no-broken-links/src/rule.ts

```diff
@@ -0,0 +1,178 @@
+            debug(`Error accessing {$absoluteUrl}. ${JSON.stringify(error)}`);
+
+            return context.report(url, element, `Broken link found (404 response)`);
+        };
+
+        /**
+         * The callback to handle success handler returned from the `head` method
+         * We will check the response status againist the brokenStatusCodes list
+         * and report if it exist there. We will also add it to the fetchedUrls
+         * so that duplicate requests will not be made if 2 links has same href value
+         */
+        const handleSuccess = (networkData: NetworkData, url: string, element: IAsyncHTMLElement) => {
+            const statusIndex = brokenStatusCodes.indexOf(
+                networkData.response.statusCode
+            );
+
+            if (statusIndex > -1) {
+                return context.report(url, element, `Broken link found (${brokenStatusCodes[statusIndex]} response)`);
+            }
+
+            fetchedUrls.push({ status: networkData.response.statusCode, url });
+
+            return Promise.resolve();
+        };
+
+        /**
+         * Checks a url againist the fetchedUrls array and return the entry if it exist
+         * The entry has 2 properties, the `url` and the `statusCode`
+         */
+        const getFetchedUrl = (url: string) => {
+
+            const filteredItems = fetchedUrls.filter((value) => {
+                return value.url === url;
+            });
+
+            if (filteredItems.length) {
+                return filteredItems[0];
+            }
+
+            return null;
+        };
+
+        /**
+         * The callback to handle when an element is visited
+         * We will get the url(href,src etc) and check if it is available online
+         * We do not need to check the items received from fetch::end::* event
+         */
+        const validateElementSrcs = async (traverseElement: ElementFound): Promise<void> => {
+            const { element, resource } = traverseElement;
+            const simpleAttributes: Array<string> = ['src', 'poster', 'data', 'href'];
+
+            const urls: Array<string> = simpleAttributes.reduce((found: Array<string>, attribute: string) => {
+                const value: string = element.getAttribute(attribute);
+
+                if (value) {
+                    found.push(value);
+                }
+
+                return found;
+            }, []);
+
+            const srcset: Array<string> = parseSrcSet(element.getAttribute('srcset'));
+
+            if (srcset.length > 0) {
+                urls.push(...srcset);
+            }
+
+            const reports: Array<Promise<void>> = urls.map(async (url) => {
+                const fullUrl = URL.resolve(resource, url);
+
+                const fetched = getFetchedUrl(fullUrl);
+
+                if (fetched) {
+                    const statusIndex = brokenStatusCodes.indexOf(fetched.statusCode);
+
+                    if (statusIndex>-1){
```

Comment:
Need some space here, probably ESLint will complain

Reply:
Will fix. But, Interestingly ES lint did not complain :)